docs/troubleshooting.md

```diff
@@ -0,0 +1,128 @@
+
+When you install an Openshift cluster, the embedded Marketplace (OperatorHub) comes with a default list of operators that can be installed onto the cluster. A number of resources need to be investigated if however you see a message like this: 
+
+```
+No OperatorHub Items found 
+
+Please check that the OperatorHub is running and that you have created a valid OperatorSource. For more information about OperatorHub, please click here. 
+```
+
+First, investigate the OperatorSources. 
+
+```
+$ oc get opsrc -n openshift-marketplace
+
+NAME                  TYPE          ENDPOINT              REGISTRY              DISPLAYNAME           PUBLISHER   STATUS      MESSAGE                                       AGE
+certified-operators   appregistry   https://quay.io/cnr   certified-operators   Certified Operators   Red Hat     Succeeded   The object has been successfully reconciled   22h
+community-operators   appregistry   https://quay.io/cnr   community-operators   Community Operators   Red Hat     Succeeded   The object has been successfully reconciled   22h
+redhat-operators      appregistry   https://quay.io/cnr   redhat-operators      Red Hat Operators     Red Hat     Succeeded   The object has been successfully reconciled   22h
+```
+
+If the status of any of the OperatorSource is anything other than `Succeeded`, you can investigate the OperatorSource with 
+`oc describe opsrc <Name-of-OperatorSource> -n openshift-marketplace`. [This section](#operatorSource-failing-to-download-from-datastore) of the doc also contains helpful information about debugging an OperatorSource. 
+
+If the OperatorSources look healthy, check that the CatalogSourceConfigs for the corresponding OperatorSources were created.
+
+```
+$ oc get csc -n openshift-marketplace
+
+NAME                  STATUS      MESSAGE                                       AGE
+certified-operators   Succeeded   The object has been successfully reconciled   23h
+community-operators   Succeeded   The object has been successfully reconciled   23h
+redhat-operators      Succeeded   The object has been successfully reconciled   23h
+```
+
+If the status of any of the CatalogSourceConfigs is anything other than `Succeeded`, you can inspect the CatalogSourceConfig with `oc describe csc <Name-of-CatalogSourceConfig>`. [This section](#catalogsourceconfig-stuck-in-configuring-phase) of the doc also contains helpful information about debugging a CatalogSourceConfig. 
+
+If the CatalogSourceConfigs look healthy, make sure that there's a pod corresponding to each CatalogSourceConfig and that they are in the `Running` state.
+
+```
+$ oc get pods -n openshift-marketplace
+
+NAME                                    READY     STATUS    RESTARTS   AGE
+certified-operators-79f876c76d-mbtjg    1/1       Running   0          86m
+community-operators-854c564f5f-9vk2x    1/1       Running   0          86m
+marketplace-operator-6dcd66cb8c-2wmkz   1/1       Running   0          86m
```

Comment:
You could curate this list of pods to only include grpc pods with a label selector as well.

Reply:
@awgreene 

Doesn't look like there's a good label to fetch all the grpc pods with. For example, here's the label of a grpc pod:
```
labels:
    marketplace.catalogSourceConfig: certified-operators
    pod-template-hash: 858975865f
```

Reply:
Agreed.